app/components/ShareModal/CopiableUrlInput.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { PropTypes } from 'react';
+
+
+class CopiableUrlInput extends React.Component {
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.setButtonEl = this.setButtonEl.bind(this);
+    this.onMouseLeave = this.onMouseLeave.bind(this);
+  }
+
+  componentDidMount() {
+    const Clipboard = require('clipboard'); // eslint-disable-line global-require
+
+    this.clipboard = new Clipboard(this.$button);
+    this.clipboard.on('success', (e) => {
+      e.trigger.setAttribute('aria-label', 'Copied!');
+      e.trigger.setAttribute('class', 'button clipboard tooltipped tooltipped-e');
+    });
+    this.$button.addEventListener('mouseleave', this.onMouseLeave);
+  }
+
+  componentWillUnmount() {
+    this.$button.removeEventListener('mouseleave', this.onMouseLeave);
+    this.clipboard.destroy();
+  }
+
+  onMouseLeave() {
+    this.$button.addEventListener('mouseleave', () => {
+      this.$button.removeAttribute('aria-label');
+      this.$button.setAttribute('class', 'button clipboard');
+    });
+  }
+
+  setButtonEl(node) {
+    this.$button = node;
+  }
+
+  render() {
+    return (
+      <div className="input-group">
+        <input
+          id={this.props.name}
+          type="url"
+          defaultValue={this.props.value}
+          className="input-group-field"
+          readOnly
+        />
+        <div className="input-group-button">
+          <button
+            ref={this.setButtonEl}
+            className="button clipboard"
+            data-clipboard-target={`#${this.props.name}`}
+          >
```

Comment:
Maybe you should try an `input[type="submit"]` here to fix this alignment bug on FF:

![monod](https://cloud.githubusercontent.com/assets/956157/17618121/3d8e62a6-607e-11e6-9a38-5707c626b11a.jpg)